### ISSUE TITLE
Semi-asynchronous replication

### ISSUE THREAD

**Issue by ghost** [hazelcast/hazelcast#92]
Some clustered DBs  have something called &quot;semi-asynchronous&quot; replication. 

Basically, in this mode, when they do something like putAsync (or put) they wait only until at least 1 of N confirmation is received from replica nodes, where N is the number of required backups. Once it is received, they proceed further. This feature obviously a tradeoff between strong consistency  and throughput. So, some people use it when they need something between a purely synchronous and a purely asynchronous replication. 

Would be nice to have the possibility to configure this option on a per distributed HZ data-structure basis using the usual config machinery. It should be possible to say that semi-async mode should be used and how many backup nodes should confirm the operation (default is 1, but it could be any K &lt;= N), before it unblocks the caller and it can proceed. 

Obviously, if K = 0, it is the async replication. And if K=N it is a sync replication. Everything is between is semi-asynchronous.

The discussion on the mailing list can be found here:
https://groups.google.com/forum/?fromgroups#!topic/hazelcast/h-fQVQggZK8

Talip liked the idea of having this feature<p>Migrated from http://code.google.com/p/hazelcast/issues/detail?id=819</p>
